cmd/infra/aws/destroy.go

```diff
@@ -78,8 +81,13 @@ func NewDestroyCommand() *cobra.Command {
 }
 
 func (o *DestroyInfraOptions) Run(ctx context.Context) error {
+	destroyInfraCtx, destroyInfraCtxCancel := context.WithTimeout(ctx, o.AwsInfraGracePeriod)
```

Comment:
you probably need to check if `AwsInfraGracePeriod` is set here and use a default if not set before using it.

Setting the default value in the args/flags is not enough as this code could be called from somewhere else like  we do in our e2e here https://github.com/openshift/hypershift/blob/main/test/e2e/util/fixture.go#L81-L87